src/pkg/cli/debug.go

```diff
@@ -74,15 +81,59 @@ func Debug(ctx context.Context, c client.FabricClient, p client.Provider, etag t
 		Services: failedServices,
 		Project:  project.Name,
 	}
-	err := p.Query(ctx, &req)
+
+	if etag != "" {
+		err := p.Query(ctx, &req)
+		if err != nil {
+			return err
+		}
+	} else {
+		return errors.New("no information to use for debugger")
+	}
+
+	resp, err := c.Debug(ctx, &req)
 	if err != nil {
 		return err
 	}
+
+	printDebugReport(resp)
+
+	// for _, request := range resp.Requests {
+	// 	term.Info(request)
+	// }
+	return nil
+}
+
+func DebugComposeFileLoadError(ctx context.Context, c client.FabricClient, p client.Provider, project *compose.Project, loadErr error) error {
+	term.Debug("Invoking AI debugger for Load Error")
+
+	files := findMatchingProjectFiles(project, nil)
```

Comment:
No point in doing a bunch of file IO if one of the exit conditions are met, so I think we should check those before looking for matching files.

```
	term.Debug("Invoking AI debugger for Load Error")
	if DoDryRun {
		return ErrDryRun
	}

	if loadErr == nil {
		return errors.New("no information to use for debugger")
	}

	req := defangv1.DebugRequest{
		Files:   files := findMatchingProjectFiles(project, nil),
		Project: project.Name,
		Logs: loadErr.Error()
	}

	resp, err := c.Debug(ctx, &req)
```